src/litserve/specs/openai.py

```diff
@@ -311,16 +311,14 @@ def _encode_response(self, output: Union[Dict[str, str], List[Dict[str, str]]])
         elif isinstance(output, dict) and "content" in output:
             message = output.copy()
             message.update(role="assistant")
-        elif self.validate_chat_message(output):
-            message = output
```

Comment:
Why did you remove it?

I see that the check above partially matches `validate_chat_message`, perhaps that why?
But in the check above you overwrite `role` anyway, even if it was already in the dict.

So maybe keep it, but move above  `elif isinstance(output, dict) and "content" in output:`